### ISSUE TITLE
CLI: Hotswap - We don't support attributes of the 'AWS::ElastiCache::CacheCluster' resource

### ISSUE THREAD

**Issue by automartin5000** [aws/aws-cdk#18809]
### What is the problem?

Can't use `cdk deploy --hotswap` with an ElastiCache CacheCluster. Get the error:
```Error: We don't support attributes of the 'AWS::ElastiCache::CacheCluster' resource. This is a CDK limitation. Please report it at https://github.com/aws/aws-cdk/issues/new/choose```
Just following the instructions here :smi

### Reproduction Steps

Try to run `cdk deploy --hotswap` on a previously deployed cdk stack with a Fargate Service/Task that has an ElastiCache endpoint in it's environment variables.

### What did you expect to happen?

Successful hotswap

### What actually happened?

Received an error
```Error: We don't support attributes of the 'AWS::ElastiCache::CacheCluster' resource. This is a CDK limitation. Please report it at https://github.com/aws/aws-cdk/issues/new/choose```

### CDK CLI Version

2.8.0

### Node.js Version

v16.13.2

### OS

Mac OS 12.1

### Language

Python

**Comment by automartin5000** [comment]
Hey Adam,

Is this helpful: `self.scope.app_env_vars["CACHE_LOCATION"] = self.memcache_cluster.attr_configuration_endpoint_address`

**Comment by skinny85** [comment]
Yes, but if you could show me how `self. memcache_cluster` is created, that would be helpful too 🙂.

**Comment by automartin5000** [comment]
@skinny85

Here you go, let me know if you need more.

```python 

        self.memcache_cluster = elasticache.CfnCacheCluster(
            self,
            "ElastiCacheCluster",
            **elasticache_params,
            engine="memcached",
            vpc_security_group_ids=[
                self.elasticache_security_group.get_att("GroupId").to_string()
            ],
            cache_subnet_group_name=self.elasticache_subnet_group.ref
        )
        self.scope.app_env_vars["CACHE_LOCATION"] = self.memcache_cluster.attr_configuration_endpoint_address
```

**Comment by skinny85** [comment]
Almost! 😄 What's `self.scope.app_env_vars`? Is it jus a simple dictionary, or something else?

**Comment by automartin5000** [comment]
Yeah just a dictionary that gets passed into the `environment` arg for the task definition

**Comment by skinny85** [comment]
Sounds good! Thanks for all of the info. I know what I need to do 😉.

**Comment by skinny85** [comment]
Actually, I may have spoken too soon 😕.

The problem is that you're using an attribute of the `AWS::ElastiCache::CacheCluster` resource (`ConfigurationEndpoint.Address`) that can't be easily constructed from the name of the Cluster, because it's a URL that's generated by the ElastiCache Service, like ` tes-ca-10pps2g12dfv.zy7new.cfg.usw2.cache.amazonaws.com`. That means it's really difficult to get the value of that attribute when hotswapping - we would have to do an ElastiCache-specific service call to do it, and we don't have a mechanism for this now.

Apologies for the bad experience @automartin5000, but it won't be easy to support hotswapping for this particular use case 😕.

I'll keep this issue open to track this gap, but it will be non-trivial to fix for sure.

**Comment by automartin5000** [comment]
Thanks @skinny85 for looking into this so quickly. What about a flag to just ignore unsupported CFN attributes and echo a warning on `deploy`?

**Comment by skinny85** [comment]
Isn't that basically what happens now though? (I understand that the output says "Error", but in fact the deployment _should_ proceed, just not a hotswap one - a full CloudFormation one. Is this not what happens to you?)

Or are you saying that you want a flag to suppress showing the warning about `AWS::ElastiCache::CacheCluster`?

**Comment by automartin5000** [comment]
<img width="835" alt="Screen Shot 2022-02-04 at 21 22 58" src="https://user-images.githubusercontent.com/7469260/152625309-65f002d2-bc66-4940-9227-91b24c106851.png">
Does not continue, just exits ^^

**Comment by skinny85** [comment]
Hmmm. That's not what I expected 😛.

What command did you execute here? Seeing the entire console output, preferably with the `--verbose` switch, would be super helpful (feel free to use GitHub Gists if it's too long).

**Comment by automartin5000** [comment]
I'd rather not dump the entire output here with IAM role names, resource IDs, and account numbers. The command I ran was: `cdk deploy --hotswap dev --app 'cdk.out/' --require-approval never`

**Comment by skinny85** [comment]
I tried it with `--app cdk.out` too, just to make sure I wasn't going crazy - same result:

```
$ yarn cdk deploy --hotswap --app cdk.out
yarn run v1.22.17
$ cdk --plugin aws-cdk-isengard deploy --hotswap --app cdk.out

✨  Synthesis time: 0.02s

⚠️ The --hotswap flag deliberately introduces CloudFormation drift to speed up deployments
⚠️ It should only be used for development - never use it for your production Stacks!
CdkWatchMinimalEcsServiceStack: deploying...
[0%] start: Publishing 06134eb41d4fad8134760d5ee6f5830f70d1e35384186ac44622b3d251fe3b20:current
[100%] success: Published 06134eb41d4fad8134760d5ee6f5830f70d1e35384186ac44622b3d251fe3b20:current
Could not perform a hotswap deployment, because the CloudFormation template could not be resolved: We don't support attributes of the 'AWS::ElastiCache::CacheCluster' resource. This is a CDK limitation. Please report it at https://github.com/aws/aws-cdk/issues/new/choose
Falling back to doing a full deployment
CdkWatchMinimalEcsServiceStack: creating CloudFormation changeset...
[··························································] (0/4)

4:59:14 PM | UPDATE_IN_PROGRESS   | AWS::CloudFormation::Stack                | CdkWatchMinimalEcsServiceStack
```
